src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/world/WorldFluidCallback.java

```diff
@@ -0,0 +1,35 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1.world;
+
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.containers.SidedFluidContainer;
+
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * this is a callback that is invoked in {@link io.github.fablabsmc.fablabs.api.fluidvolume.v1.FluidView}.
+ * when a mod attempts to find a fluid container in the world, here, mods can add custom handling for when
+ * blocks aren't sufficient for whatever reason.
+ */
+public interface WorldFluidCallback {
+	Event<WorldFluidCallback> EVENT = EventFactory.createArrayBacked(WorldFluidCallback.class, c -> (w, b) -> {
```

Comment:
LBA, Universal Components, or Fluidity adding their own compat hooks in.

Reply:
For the unfortunate event in which someone decides that it would be a good idea to have infinite water sources not bound to blocks or entities.

It's just a general purpose thing, I could remove it

Reply:
Actually now that I think about it, the event is pretty useless